src/transformers/models/maskformer/modeling_maskformer.py

```diff
@@ -0,0 +1,1638 @@
+
+        Args:
+            feature_size (int, optional): The features (channels) of FPN feature maps. Defaults to 256.
+            mask_feature_size (int, optional): The features (channels) of the target masks size $C_{\epsilon}$ in the paper. Defaults to 256.
+        """
+        super().__init__()
+        self.fpn = FPNModel(*args, feature_size=feature_size, **kwargs)
+        self.mask_proj = nn.Conv2d(feature_size, mask_feature_size, kernel_size=3, padding=1)
+
+    def forward(self, features: List[Tensor]) -> Tensor:
+        fpn_features: List[Tensor] = self.fpn(features)
+        # we use the last feature map
+        x = self.mask_proj(fpn_features[-1])
+        return x
+
+
+# copied from original implementation, also practically equal to DetrSinePositionEmbedding
+class PositionEmbeddingSine(nn.Module):
+    """
+    This is a more standard version of the position embedding, very similar to the one
+    used by the Attention is all you need paper, generalized to work on images.
+    """
+
+    def __init__(
+        self, num_pos_feats: int = 64, temperature: int = 10000, normalize: bool = False, scale: Optional[float] = None
+    ):
+        super().__init__()
+        self.num_pos_feats = num_pos_feats
+        self.temperature = temperature
+        self.normalize = normalize
+        if scale is not None and normalize is False:
+            raise ValueError("normalize should be True if scale is passed")
+        if scale is None:
+            scale = 2 * torch.pi
+        self.scale = scale
+
+    def forward(self, x: Tensor, mask: Optional[Tensor] = None) -> Tensor:
+        if mask is None:
+            mask = torch.zeros((x.size(0), x.size(2), x.size(3)), device=x.device, dtype=torch.bool)
+        not_mask = ~mask
+        y_embed = not_mask.cumsum(1, dtype=torch.float32)
+        x_embed = not_mask.cumsum(2, dtype=torch.float32)
+        if self.normalize:
+            eps = 1e-6
+            y_embed = y_embed / (y_embed[:, -1:, :] + eps) * self.scale
+            x_embed = x_embed / (x_embed[:, :, -1:] + eps) * self.scale
+
+        dim_t = torch.arange(self.num_pos_feats, dtype=torch.float32, device=x.device)
+        dim_t = self.temperature ** (2 * torch.div(dim_t, 2, rounding_mode="floor") / self.num_pos_feats)
+
+        pos_x = x_embed[:, :, :, None] / dim_t
+        pos_y = y_embed[:, :, :, None] / dim_t
+        pos_x = torch.stack((pos_x[:, :, :, 0::2].sin(), pos_x[:, :, :, 1::2].cos()), dim=4).flatten(3)
+        pos_y = torch.stack((pos_y[:, :, :, 0::2].sin(), pos_y[:, :, :, 1::2].cos()), dim=4).flatten(3)
+        pos = torch.cat((pos_y, pos_x), dim=3).permute(0, 3, 1, 2)
+        return pos
+
+
+class MLP(nn.Sequential):
```

Comment:
Isn't this the second `MLP`in this file?

Reply:
This is the MLP used on top of the decoder outputs, I assume. In DETR, I called this class `DetrMLPPredictionHead`.
So here we could call it `MaskformerMLPPredictionHead`, maybe?